amaranth/vendor/_gatemate.py

```diff
@@ -0,0 +1,120 @@
+from abc import abstractmethod
+from amaranth import *
+from amaranth.build import *
+from amaranth.lib.cdc import ResetSynchronizer
+
+__all__ = ["GateMatePlatform"]
+
+class GateMatePlatform(TemplatedPlatform):
+    """
+    .. rubric:: GateMate FPGA toolchain
+
+    Required tools:
+        * ``yosys``
+        * ``p_r``
+
+    The environment is populated by setting the ``CC_TOOL`` environment variable to point to the toolchain directory.
```

Comment:
I don't see how this would happen as nothing in this file touches `CC_TOOL`. There is a default behavior that will use the `platform.toolchain` property (defined here to be `GateMate`) to derive `AMARANTH_ENV_GATEMATE` from it.

If you wanted to use `CC_TOOL` to populale the environment you have to write a custom template for the shell and batch scripts, which is possible but finicky and requires extensive testing and especially careful coding.

Reply:
Changed `CC_TOOL` to `AMARANTH_ENV_GATEMATE`